underscore.mdx

```diff
@@ -5,133 +5,146 @@ description: Using underscore expressions to define features
+With our `Transaction` example, we can compute the number of large transactions a user as had as well as their total
+spending volume:
 
 ```diff-py
-- from chalk.features import features, has_one, has_many, DataFrame
-+ from chalk.features import _, features, has_one, has_many, DataFrame
+from chalk.features import _, features, DataFrame
 
 @features
-class Astronaut:
+class Transaction:
     id: int
-    name: str
-    height: float
-    mass: float
-    ship_id: int
+    user_id: "User.id"
+    total: float
+    sales_tax: float
+    subtotal: float = _.total - _.sales_tax
 
 @features
-class RocketShip:
+class User:
     id: int
-    engine_id: int
-
-    # `has_one` relationship with `RocketEngine`
-    engine: RocketEngine = has_one(lambda: RocketShip.engine_id == RocketEngine.id)
-
-    # `has_many` relationship with `Astronaut`
-    astronauts: DataFrame[Astronaut] = has_many(lambda: Astronaut.ship_id == RocketShip.id)
-
-+   # Defining total_mass using an underscore expression
-+   total_mass: float = _.engine.mass + _.astronauts[_.mass].sum()
+    # implicit has-many relationship with Transaction due to `user_id` above
+    transactions: DataFrame[Transaction]
+    large_transactions: DataFrame[Transaction] = _.transactions[_.total > 1000]
++   total_spend: float = _.transactions[_.total].sum()
++   num_large_transactions: int = _.large_transactions.count()
 ```
 
-Here we used the underscore syntax do a [projection](/docs/dataframe#projections) of the `mass` column in the
-astronauts `DataFrame`.
-
-
-### Filters
-
-Underscores can also be used to define [filters](/docs/dataframe#filters) on `DataFrame` features.
-
-For example, suppose we wanted to define a feature called `count_tall_astronauts` on `RocketShip`.
-We could define the feature as follows
-
-```diff-py
-@features
-class RocketShip:
-    id: int
-    engine_id: int
-
-    # `has_one` relationship with `RocketEngine`
-    engine: RocketEngine = has_one(lambda: RocketShip.engine_id == RocketEngine.id)
+To compute `User.total_spend`, we needed to create a projection of the `User.transactions` DataFrame limited to only the
+`Transaction.total` column so that the `sum` aggregation could work. In contrast, no projection was needed for
+`num_large_transaction`'s `count` aggregation because `count` works on DataFrames with any number of columns.
+
+## All supported operations
+
+### Arithmetic
+* Addition: `+`
+* Subtraction: `-`
+* Multiplication: `*`
+* Division: `/`
+
+### Conditions
+* Greater than: `>`
+* Greater than or equal: `>=`
+* Less than: `<`
+* Less than or equal: `<=`
+* Equal: `==`
+* Not equal: `!=`
+* Boolean and: `&`
+* Boolean or: `|`
+
+Be careful to not use Python's `and`, `or`, `not`, or `is` operators. Python does not allow these operators to be
+overridden. They take precedence and will be evaluated with native Python, so they will not work with Chalk's underscore
+expressions.
+
+### Aggregations
+* `sum`
```

Comment:
It could be helpful to specify some of the edge cases around these operators. e.g. something like

- sum
  - `null` values are skipped while computing `.sum()`
  - if the dataframe is empty or all of its values are `null`, then `0` is returned

- min
  - `null` values are skipped while computing `.min()`
  - if the dataframe is empty or all of its values are `null`, then `null` is returned. Ensure that the corresponding feature is marked as optional to handle this case, or use `_.coalesce` to provide a default value when the `.min()` is `null` 


Also, `approx_percentile` takes a parameter e.g. `approx_percentile(0.75)` returns a value which is approximately equal to the 75% percentile of the values in the dataframe